include/linux/soundwire/sdw_intel.h

```diff
@@ -16,24 +16,85 @@ struct sdw_intel_ops {
 };
 
 /**
- * struct sdw_intel_res - Soundwire Intel resource structure
+ * struct sdw_intel_acpi_info - Soundwire Intel information found in ACPI tables
+ * @handle: ACPI controller handle
+ * @count: link count found with "sdw-master-count" property
+ * @link_mask: bit-wise mask listing links enabled by BIOS menu
+ *
+ * this structure could be expanded to e.g. provide all the _ADR
+ * information in case the link_mask is not sufficient to identify
+ * platform capabilities.
+ */
+struct sdw_intel_acpi_info {
+	acpi_handle handle;
+	int count;
+	u32 link_mask;
+};
+
+/**
+ * struct sdw_intel_res - Soundwire Intel global resource structure,
+ * typically populated by the DSP driver
+ *
+ * @count: link count (may be filtered by DSP driver)
```

Comment:
@plbossart I am not sure if I understand what dose "may be filtered by DSP driver" means. Does it means we may only want to use less than the amount of links? But it is always start from link 0, right? The word "filtered" sounds link we can mask some of the links off. Like what link_mask do.

Reply:
good point. I meant that the DSP driver is under no obligation to use all links and can decide to ignore some links.